### PR TITLE
fix issues for P007, thanks, wf!

### PR DESCRIPTION
**Pull request by updownlife**

### CODE REVIEW

File: hudeven/P007.py [hunk 1]
```diff
@@ -4,13 +4,15 @@ def reverse(self, x):
...
+        MIN_INT = -MAX_INT - 1
+        if x == MIN_INT: return 0
+        
+        res = 0
         p = x if x >= 0 else -x
         while p != 0:
             digit = p % 10
-            if res < MAX_INT / 10 or ( res == MAX_INT / 10 and digit <= MAX_INT % 10):
+            if res <= MAX_INT / 10:
                 res = res * 10 + digit
```
Comment (lionelfeng) [inline hudeven/P007.py]: what if `digit = 9`

Comment (updownlife) [inline hudeven/P007.py]: For 32-bit Integer, digit will not equal to 9. digit <= 1. I proved it in your pull request.

Comment (lionelfeng) [inline hudeven/P007.py]: fair enough, but the leetcode oj does use overflowed int as the input.

**Comment by lionelfeng** [comment]
lgtm

**Pull request merged**